app/lib/dartdoc/handlers.dart

```diff
@@ -65,3 +67,51 @@ Future<shelf.Response> packageHandler(shelf.Request request) async {
 
   return notFoundHandler(request);
 }
+
+/// Handles requests for:
+///   - /documentation/<package>
+///   - /documentation/<package>/<version>
+Future<shelf.Response> documentationHandler(shelf.Request request) async {
+  final docFilePath = parseRequestUri(request.requestedUri);
+  if (docFilePath == null) {
+    return notFoundHandler(request);
+  }
+  final String requestMethod = request.method?.toUpperCase();
+  if (requestMethod == 'GET') {
+    // placeholder response until proper implementation is done
+    return jsonResponse({
+      'package': docFilePath.package,
+      'version': docFilePath.version,
+      'path': docFilePath.path,
+    });
+  }
+  return notFoundHandler(request);
+}
+
+class DocFilePath {
+  final String package;
+  final String version;
+  final String path;
+
+  DocFilePath(this.package, this.version, this.path);
+}
+
+DocFilePath parseRequestUri(Uri uri) {
+  final pathSegments = uri.pathSegments;
+  if (pathSegments.length < 3) {
+    return null;
+  }
+  if (pathSegments[0] != 'documentation') {
+    return null;
+  }
+  final String package = pathSegments[1];
+  final String version = pathSegments[2];
```

Comment:
The comment on `documentationHandler` claims we also support `/documentation/<package>`, so shouldn't `version` be optional?

Reply:
on the current dartdoc site there is no redirect from `/documentation/<package>` to `/documentation/<package>/<latest_version>`, and with that, the version is not optional. Fixed the documentation, and later we can add the redirect.